tests/e2e/common-operator-integ-suite.sh

```diff
@@ -234,10 +234,27 @@ if [ "${SKIP_BUILD}" == "false" ]; then
     # Install OLM in the cluster because it's not available by default in kind.
     ${OPERATOR_SDK} olm install
 
+    # Wait to operatorhub pod to be running
+    ${COMMAND} wait --for=condition=ready pod -l olm.catalogSource=operatorhubio-catalog -n olm --timeout=5m
```

Comment:
@fjglira You need to change this. You should wait for the CatalogSource, not the underlying pod(s). 

Waiting for the pods is the wrong thing to do and won't work. The reason this is failing is because at the exact time when you run this `wait` command, there are *two* pods (one is terminating, the other is starting up). The wait command will now wait for *both* of these pods to become ready. But of course, the one that's terminating will never become ready and thus the `wait` command will fail.

Reply:
Basically, when you use a label selector in the `kubectl wait` command, you're not telling `kubectl` to wait for at least one pod matching the selector to become ready. You're telling it to find all the pods that match the selector, and then wait for all of them to become ready.

Reply:
Changed